index.js

```diff
@@ -55,3 +64,14 @@ const endWhenStreamsDone = async (passThroughStream, streams) => {
 		}
 	}
 };
+
+const onMergedStreamEnd = async (passThroughStream, {signal}) => {
+	try {
+		await finished(passThroughStream, {signal, cleanup: true});
+	} catch {}
+};
```

Comment:
The purpose of this function is only to resolve when `passThroughStream` ends, aborts or errors. When it does, `Promise.race()` resolves, which calls `abortController.abort()`, which cleans up the other `finished()` calls.